### ISSUE TITLE
Twitter Integration (and more)

### ISSUE THREAD

**Issue by willzeng** [willzeng/cograph#468]
@davidfurlong We could build a growth strategy that leverages twitter slickly.  Here's the idea:

Users are initially prompted (and highly encouraged) to log-in with a twitter account. Then, whenever they use an @ mention in a node or a connection description on a public cograph, it will autocomplete based on who they follow on twitter and tweet a summary (with link) of the node/conn to that person. (this feature is on by default, but could be turned off in settings)

On signup there will still be an 'opt-out' option to log-in with a cograph only account.  However for these, mentions will be disabled.  If you later connect a twitter account to that cograph account, mentions will be enabled.

Pros
- Grassroots advertising at its best.  One of the easiest things to in a cograph becomes sharing it word of mouth with your friends and followers. This allows people to notify non-cograph users about what they are doing.
- We don't have to write an internal notification system.  Though, we could write one later that just duplicates the notifications in the cograph account affiliated with that twitter (should there be one).
- We don't have to deal with the redundant process of having people, once again, go through the process of selecting who to follow on a new social platform.

Cons
- There aren't notifications for non-twitter connected accounts.
- Giving the option of having people opt out of twitter integration, means that they could block your twitter username with a non-connected account.  This would be bad. We never want to prevent signups.

Reflective Pros
- This setup makes more obvious some candidates for 'Premium features', such as
  - collaborators on private graphdocs that can be privately mentioned
  - basically we treat them similarly to how github treats private repos

tl;dr  Growth is the most important thing, thus we should integrate with twitter. The problem is username blocking if we allow non-twitter signups as well.

Thoughts?

**Comment by almereyda** [comment]
There is also the issue of yet non-integrated, cited Tweets. We'd expect to see those networks similary regular RTs.
